src/util.rs

```
@@ -91,36 +90,24 @@ pub(crate) fn format_secret(
 }
 
 pub(crate) fn exec_prompt(
-    conn: zbus::Connection,
+    conn: zbus::blocking::Connection,
     prompt: &ObjectPath,
 ) -> Result<zvariant::OwnedValue> {
-    let prompt_proxy = PromptProxy::new_for(&conn, SS_DBUS_NAME, prompt)?;
+    let prompt_proxy = PromptProxyBlocking::builder(&conn)
+        .destination(SS_DBUS_NAME)?
+        .path(prompt)?
+        .build()?;
 
-    let (tx, rx) = channel();
+    let mut receive_completed_iter = prompt_proxy.receive_completed().unwrap();
```

Can this panic? If it can, we should return an error instead.